demo/test/src/components/DropdownMultiselect.js

```diff
@@ -0,0 +1,235 @@
+    if (showDropdown !== false &&
+      node.current.target == ev.target
+    ) {
+      setShowDropdown(false);
+    }
+  }
+  const handleClick = () => {
+    setShowDropdown(!showDropdown);
+  }
+  const handleChange = (ev) =>  {
+    if(ev.currentTarget === undefined) {
+      return;
+    }
+    
+    var currentSelected = [...selected];
+    let value = ev.currentTarget.value;
+    
+    if (ev.currentTarget.checked) {
+      currentSelected.push(value);
+    } else {
+      var index = currentSelected.indexOf(value);
+      currentSelected.splice(index, 1);
+    }
+
+    // update the state with the new array of options
+    setSelected(currentSelected);
+   
+    if(props.handleOnChange !== undefined) {
+      props.handleOnChange(currentSelected)
+    }
+  }
+  const getPlaceholderValue = () => {
+    if (selected.length == 0) {
+      return props.placeholder;
+    }
+
+    if (
+      props.placeholderMultipleChecked !== null &&
+      selected.length > 1
+    ) {
+      return props.placeholderMultipleChecked;
+    } else {
+      let currentOptions = [...options];
+
+      if(currentOptions.length == 0) {
+        return props.placeholder;
+      }
+
+      let selectedLabels = [];
+      let optionKey = props.optionKey;
+      
+      selected.map((row) => {
+        let foundOption = currentOptions.find((option) => {
+          return option['key'] == row;
+        });
+
+        selectedLabels.push(foundOption.label);
+      });
+
+      return selectedLabels.join(", ");
+    }
+  }
+  const handleSelectDeselectAll = () => {
+    if (selected.length == options.length) {
+      setSelected([]);
+      if(props.handleOnChange) {
+        props.handleOnChange([]);
+      }
+    } else {
+      let allOptions = options;
+
+      let newSelected = [];
+
+      allOptions.map((obj) => {
+        newSelected.push(obj.key.toString());
+      });
+
+      setSelected(newSelected);
+      if(props.handleChange) {
+        props.handleOnChange(newSelected)
+      }
+    }
+  }
+
+  useEffect(() => {
+
+    console.log(JSON.stringify(props));
+    //setOptions();
+      document.addEventListener("mousedown", handleClickOutside);
+
+      return function cleanup() {
+        document.removeEventListener(
+          "mousedown",
+          handleClickOutside
+        );
+      }
+  }, [])
+
+  useEffect(() => {
+    if(!props)
+      return;
+
+    if (props.options.length == 0) {
+      console.error("React Dropdown Multiselect Error: options array is empty.");
+      return;
+    }
+
+    let optionsArray = [];
+    if (typeof props.options[0] === "object") {
+      props.options.map((value, index) => {
+        let key = value[props.optionKey];
+        let label = value[props.optionLabel];
+
+        optionsArray.push({ key: key, label: label });
+      });
+    } else if (typeof props.options[0] === "string") {
+      props.options.map((value) => {
+        optionsArray.push({ key: value, label: value });
+      });
+    }
+
```

Comment:
to fix the pre-selected options

if (typeof props.selected[0] === "string") {
      setSelected(props.selected);      
    }

Reply:
@efillman Ty. I will update the revision and do another pull request.